server/src/main/java/io/druid/query/extraction/LookupExtractionModule.java

```diff
@@ -0,0 +1,211 @@
+              throws Exception
+          {
+            final Map<String, LookupExtractorFactory> failedUpdates = new HashMap<>();
+            for (final String name : lookups.keySet()) {
+              final LookupExtractorFactory factory = lookups.get(name);
+              try {
+                if (!manager.updateIfNew(name, factory)) {
+                  failedUpdates.put(name, factory);
+                }
+              }
+              catch (ISE ise) {
+                LOG.error(ise, "Error starting [%s]: [%s]", name, factory);
+                failedUpdates.put(name, factory);
+              }
+            }
+            return ImmutableMap.of("status", "accepted", LookupExtractionModule.FAILED_UPDATES_KEY, failedUpdates);
+          }
+
+          @Override
+          public Object get(String id)
+          {
+            return manager.get(id);
+          }
+
+          @Override
+          public Map<String, LookupExtractorFactory> getAll()
+          {
+            return manager.getAll();
+          }
+
+          @Override
+          public Object delete(String id)
+          {
+            // Prevent races to 404 vs 500 between concurrent delete requests
+            synchronized (deleteLock) {
+              if (manager.get(id) == null) {
+                return null;
+              }
+              if (!manager.remove(id)) {
+                // We don't have more information at this point.
+                throw new RE("Could not remove lookup [%s]", id);
+              }
+              return id;
+            }
+          }
+        }
+    );
+  }
+}
+
+class LookupResourceListenerAnnouncer extends ListenerResourceAnnouncer
+{
+  @Inject
+  public LookupResourceListenerAnnouncer(
+      Announcer announcer,
+      LookupListeningAnnouncerConfig lookupListeningAnnouncerConfig,
+      @Self DruidNode node
+  )
+  {
+    super(
+        announcer,
+        lookupListeningAnnouncerConfig,
+        lookupListeningAnnouncerConfig.getLookupKey(),
+        HostAndPort.fromString(node.getHostAndPort())
+    );
+  }
+}
+
+class LookupListeningAnnouncerConfig extends ListeningAnnouncerConfig
+{
+  public static final String DEFAULT_TIER = "__default";
+  @JsonProperty("lookupTier")
+  private String lookupTier = null;
+
+  @JsonCreator
+  public static LookupListeningAnnouncerConfig createLookupListeningAnnouncerConfig(
+      @JacksonInject ZkPathsConfig zkPathsConfig,
+      @JsonProperty("lookupTier") String lookupTier
+  )
+  {
+    final LookupListeningAnnouncerConfig lookupListeningAnnouncerConfig = new LookupListeningAnnouncerConfig(
+        zkPathsConfig);
+    lookupListeningAnnouncerConfig.lookupTier = lookupTier;
+    return lookupListeningAnnouncerConfig;
+  }
+
+  @Inject
+  public LookupListeningAnnouncerConfig(ZkPathsConfig zkPathsConfig)
+  {
+    super(zkPathsConfig);
+  }
+
+  public String getLookupTier()
+  {
+    return lookupTier == null ? DEFAULT_TIER : lookupTier;
```

Comment:
Good question. Because the json configurator doesn't work that way. Things have to be lazily evaluated against null (see ZkPathsConfig for example)